apps wc: stuck ingesting ingress-nginx logs

robinelastisys opened the pull request:
**What this PR does / why we need it**:
https://github.com/elastisys/compliantkubernetes-apps/issues/1317

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #1317

**Public facing documentation PR** *(if applicable)*
<!-- https://github.com/elastisys/compliantkubernetes/pull/ -->

**Special notes for reviewer**:
This is a solution that at least seems to work. Is it perfect? Not sure. Removing the limits of the pod may not be needed to stop the error from happening in the first place but it seems to be needed to get rid of it in practice when it have been going on for a long time.

**Add a screenshot or an example to illustrate the proposed solution:**
![Screenshot from 2023-01-02 09-57-35](https://user-images.githubusercontent.com/80628178/210210956-69d4a753-bbfa-488a-a2e4-5430d06d428e.png)

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [x] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [x] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

<!--
Here are the commit prefixes and comments on when to use them:
all: (things that touch on more than one of the areas below, or don't fit any of them)
apps: (changes to the applications running in both/all clusters)
apps sc: (changes to applications in the service cluster)
apps wc: (changes to applications in the workload cluster)
docs: (documentation)
tests: (test related changes)
pipeline: (the pipeline)
config: (configuration, e.g. add/remove/rename a parameter, this is not for changes to the default values for an application that would go into `apps [sc/wc]`)
bin: (changes to binaries or scripts used manage ck8s)
release: (anything release related)

Example commit prefix usage:

git commit -m "docs: Add instructions for how to do x"
-->

robinAwallace commented on helmfile/values/fluentd-user.yaml.gotmpl:
```diff
@@ -206,6 +206,14 @@ extraConfigMaps:
+        key message
+        pattern /\[error\]\s[0-9]+#[0-9]+:\s\*([A-Za-z0-9]+( [A-Za-z0-9]+)+):/
+      </exclude>
+    </filter>
+
```
Could you add another filter to remove any messages containing binary symbols, `\\xXY` XY => 0-9 A-F, from `kubernetes.var.log.containers.ingress-nginx-controller-**`

crssnd commented on config/config/wc-config.yaml:
```diff
@@ -256,7 +256,6 @@ fluentd:
   user:
     resources:
       limits:
-        cpu: 500m
```
I do not think we want this. as fluentd can use the entire cpu and affected other pods and services

Xartos commented on config/config/wc-config.yaml:
```diff
@@ -254,6 +254,7 @@ fluentd:
   ## Users can specify additional plugins and config in the respective configmaps:
   ## 'fluentd-extra-plugins', and 'fluentd-extra-config'.
   user:
+
```
```suggestion
```

Xartos commented on WIP-CHANGELOG.md:
```diff
@@ -16,6 +16,7 @@
 - Fixed a templating error which occurs when more than one workload cluster is specified under the `global.clustersMonitoring` in the `sc-config.yaml`
 - Fix `KubeletDown` alert rule, did previously not alert if a kubelet was missing.
 - Add permissions to the `alerting_full_access` role in Opensearch to be able to view notification channels.
+- Added a fluentd <exclude> pattern to our ingress-nginx-controller-containers to prevent them from overloading and `fluent-plugin-record-modifier` was added to our image `ghcr.io/elastisys/fluentd:v3.4.0-ck8s5` to prevent further errors from happening.
```
The `<exclude>` thing doesn't apply any more right?

robinelastisys replied:
Yeah it shouldn't be needed if the new plugin fixes it all :thinking_face:

crssnd commented on helmfile/values/fluentd-user.yaml.gotmpl:
```diff
@@ -130,6 +130,11 @@ extraConfigMaps:
       </parse>
     </source>
 
+    <filter **>
```
shouldn't we add this filter for fluentd-wc and sc configs?

robinelastisys replied:
We do not need it in sc, but yeah, i add it in wc also now ofc! :+1:

crssnd commented:
I did a quick test and I get this error:
` Failed to pull image "ghcr.io/elastisys/fluentd:v3.4.0-ck8s5": rpc error: code = Unknown desc = failed to pull and unpack image "ghcr.io/elastisys/fluentd:v3.4.0-ck8s5": failed to resolve reference "ghcr.io/elastisys/fluentd:v3.4.0-ck8s5": failed to authorize: failed to fetch anonymous token: unexpected status: 401 Unauthorized`

please test this changes so we do not create additional issues

robinelastisys commented:
> I did a quick test and I get this error: ` Failed to pull image "ghcr.io/elastisys/fluentd:v3.4.0-ck8s5": rpc error: code = Unknown desc = failed to pull and unpack image "ghcr.io/elastisys/fluentd:v3.4.0-ck8s5": failed to resolve reference "ghcr.io/elastisys/fluentd:v3.4.0-ck8s5": failed to authorize: failed to fetch anonymous token: unexpected status: 401 Unauthorized`
> 
> please test this changes so we do not create additional issues

Just had to make the registry repo public. Should work now.

robinAwallace commented:
Just a general reminder, do commits than force push when addressing comments and do a squash in the end. Because I do not remember how it was originally anymore.

> shouldn't we add this filter for fluentd-wc and sc configs?

Yes would be good

robinelastisys commented:
> Just a general reminder, do commits than force push when addressing comments and do a squash in the end. Because I do not remember how it was originally anymore.
> 
> > shouldn't we add this filter for fluentd-wc and sc configs?
> 
> Yes would be good

yeah, gets a bit much after what feels like 50 commits :+1:

crssnd reviewed (approved):
did a quick test and didn't find any issues

robinAwallace reviewed (approved):
LGTM, nicely done :+1:
